src/io-methods/filesystem.js

```diff
@@ -1,47 +1,17 @@
 "use strict";
-const fs = require("fs");
-const config = require("../config").config;
+const fs = require("fs/promises");
+const { getConfig } = require("../config");
 const jsHelpers = require("./js-file-helpers.js");
 
 exports.readManifest = function (filePath) {
-  return new Promise((resolve, reject) => {
-    //create file if not already created
-    fs.open(filePath, "a", function (err, fd) {
-      if (err) reject(`Could not open file ${filePath}`);
-      else {
-        fs.readFile(filePath, "utf8", function (err2, data) {
-          if (err2) {
-            console.error(err2);
-            reject(`Could not read file ${filePath}`);
-          } else resolve(data);
-        });
-      }
-    });
-  });
+  return fs.readFile(filePath, "utf-8");
 };
 
 exports.writeManifest = function (filePath, data) {
-  const jsonPromise = new Promise((resolve, reject) => {
-    fs.writeFile(filePath, data, function (err) {
-      if (err) reject(`Could not write file ${filePath}`);
-      else resolve();
-    });
-  });
+  const config = getConfig();
+  const useJsHelpers = config && config.writeJsFile;
+  const finalFilePath = useJsHelpers ? jsHelpers.getJsPath(filePath) : filePath;
+  const finalData = useJsHelpers ? jsHelpers.createJsString(data) : data;
 
-  const jsPromise = new Promise((resolve, reject) => {
-    if (!config || !config.writeJsFile) {
-      resolve();
-    } else {
-      fs.writeFile(
-        jsHelpers.getJsPath(filePath),
-        jsHelpers.createJsString(data),
-        function (err) {
-          if (err) reject(`Could not write file ${filePath}`);
-          else resolve();
-        }
-      );
-    }
-  });
-
-  return Promise.all([jsonPromise, jsPromise]);
```

Comment:
I just simplified the implementation approach here. The Promise.all was a race condition as it was potentially writing the file twice, depending on configuration.